BLSQ/iaso
# HOTFIX: potential payments race condition

Explain what problem this PR is resolving

Related JIRA tickets : IA-3612

## Self proofreading checklist


## Doc
- Tell us where the doc can be found (docs folder, wiki, in the code...)

## Changes

- Add unique constraint on `PotentialPayment`'s `user`
- In `calculate_potential_payments`, wrap the block of code calling `get_or_create` and `save` in a `try/except`. Return existing `PotentialPayment` in case of `IntegrityError`
- Test the unicity constraint on the model

## How to test

In a DB with at leats one potential payment, go to django admin and try to create a potential payment for a user that already has one. You should get an error

## Print screen / video


https://github.com/user-attachments/assets/bbc9e797-c802-4caf-8e0f-f9384c518f70



## Notes

Can be hotfixed on v1.246
